test/test-applications/integrations/TestApplication.Wcf.Server.NetFramework/StatusService.cs

```diff
@@ -35,15 +35,20 @@ public class StatusService : IStatusServiceContract
 
     public Task<StatusResponse> PingAsync(StatusRequest request)
     {
-        lock (Lock)
+        var time = DateTimeOffset.UtcNow;
+
+        try
         {
-            TimesHit += 1;
+            return Task.FromResult(
+                new StatusResponse { ServerTime = time, });
         }
-
-        return Task.FromResult(
-            new StatusResponse
+        finally
+        {
+            lock (Lock)
             {
-                ServerTime = DateTimeOffset.UtcNow,
-            });
+                TimesHit += 1;
```

Comment:
I see two options
1. Use a `Semaphore` for proper synchronization
2. Run the server app using `StartTestApplication`+kill at the end of the test instead of `RunTestApplication`

Reply:
The server app is already run using its own helper not `RunTestApplication` from `TestHelper`

Reply:
Ad the test will kill in `Dispose()` so this code does not seem needed anyway.